go/bindings/build.go

```diff
@@ -64,6 +65,13 @@ type BuildArgs struct {
 	MaterializeDriverFn MaterializeDriverFn
 }
 
+// We currently run all validations of a catalog spec from a developer's local
```

Comment:
I think this should be inverted: we should model a concept of the current build/run context, probably as a protobuf enum (name: `pf.Invocation` ? ). We'd then thread that through and make a decision about host networking at the last possible mile, within the guts of the docker connector implementation.

That enum definition can be very literal & tightly coupled to the specific modes of operations we have now ("TEST" vs "DEVELOP" vs "SERVE" for now, perhaps), rather than those we hope to have. Part of the benefit of this factoring is that, when we add / remove modes of operation, those become compile-time errors which are easy to root out & update.